### ISSUE TITLE
Feature request/suggestion - Duplicate Nick Names

### ISSUE THREAD

**Issue by impmallet** [EssentialsX/Essentials#1822]
Hello,

I'd like to request an old feature be re-implemented in essentialsx. In older versions of essentials it was possible to issue duplicate nick names to players, but at some point this was removed (most likely because of how easily it can be abused)

Given that this ability would not be suitable for all servers, I suggest adding a configuration line giving server operators the choice  of removing the duplicate name check, such as:

"allow-duplicate-nicks: true"

Thank you.

EssentialsX version (run `/essentials`): 

Server software (run `/version`): 

Server (`logs/latest.log`):

EssentialsX config (`plugins/Essentials/config.yml`): 

Details:

**Comment by DrewAMitchell** [comment]
I'm not sure if we did act on this, but I remember this issue being presented, stating that `/realname` may not return the right one or something to that effect...
Ah, [here it is](https://github.com/EssentialsX/Essentials/issues/1649). Doesn't look like they've changed this but I could be wrong...?

**Comment by impmallet** [comment]
Hello!

Indeed, someone must have changed something, as currently you can't have 2
identical nick names. I did discover that if you manually add a nick to a
player config file, you can still force any nick you want, but of course
that requires searching through hundreds of player files all named with
UUID.

It would be fantastic if the name check could be toggled on or off in the
config, as I really loved being able to swap player names around/duplicate
them. It creates some interesting chat conversations :P

**Comment by mdcfe** [comment]
This has been the behaviour for [as far back as Essentials' version history goes](https://github.com/EssentialsX/Essentials/blame/03fd6249fc61f15d7e153ae66bccc77b0dc36447/Essentials/src/com/earth2me/essentials/commands/Commandnick.java#L84). There may have been side effects that have influenced how the check works, but it's been part of Essentials for as long as anyone can remember.

I'm not sure that there actually is a case for us incorporating this into EssentialsX, since it doesn't add any value to the plugin nor does it bring any kind of benefit to the vast majority of servers.

**Comment by impmallet** [comment]
I can assure you that it worked on prior versions, and have evidence to
prove it.

I looked over your git history and it seems you have a habit of just rapid
fire closing issues. You personally do not feel this feature has value, but
you have a user of your plugin that is literally vouching for it.

Thank you for showing me how your dev team operates, I will use a different
set of plugins going forwars and will stop recommending essentialsx to
other operators.

On Feb 4, 2018 9:52 AM, "md678685" <notifications@github.com> wrote:

This has been the behaviour for as far back as Essentials' version history
goes
<https://github.com/EssentialsX/Essentials/blame/03fd6249fc61f15d7e153ae66bccc77b0dc36447/Essentials/src/com/earth2me/essentials/commands/Commandnick.java#L84>.
There may have been side effects that have influenced how the check works,
but it's been part of Essentials for as long as anyone can remember.

I'm not sure that there actually is a case for us incorporating this into
EssentialsX, since it doesn't add any value to the plugin nor does it bring
any kind of benefit to the vast majority of servers.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/EssentialsX/Essentials/issues/1822#issuecomment-362912214>,
or mute the thread

**Comment by Kakifrucht** [comment]
Welcome to EssentialsX, an open source project where @md678685 is a volunteer developing it in his free time. Have a nice sunday :)

Thanks @md678685 for cleaning up the mess that was the issue section for this project!

**Comment by mdcfe** [comment]
If you could confirm that it has definitely worked in the past, feel free - I'm curious as to why the behaviour would change, but the fact remains that this has been an Essentials feature for over 6 years.

---

The aim of both the original Essentials and of EssentialsX is to provide the "essential" features for a server. Over the few years that EssentialsX has existed, we've received many, *many* requests to add new features to the plugin. On some occasions, these feature requests add real value to the thousands of servers that use EssentialsX, and the evidence is usually that we receive numerous requests for the feature and people upvote and discuss these features, either in our Discord or in the issue tracker. However, we also receive many feature requests that are not "essential" to most servers, and are therefore not within the scope of the project.

The other problem is that with every new feature that gets added, we then need to fully evaluate the purpose of the feature, decide how to implement it, write the code for it, test the newly-added code and document it, as well as potentially responding to questions and concerns raised by the addition of the feature. This is only amplified by the complexity of the EssentialsX codebase, which further increases with every addition to the plugin, and a surprising number of people are still concerned by how large the EssentialsX jar actually is, since it includes so many features out of the box. Individual features not only take time initially to implement, but then consume our time when supporting the entire plugin.

If that doesn't seem like a lot to you, remember that we don't maintain this plugin as our job. It's something we contribute our free time towards, in order to benefit the wider Minecraft server community. Adding new features and changing behaviour constantly means that our free time would become filled with trying to support these changes that may only benefit 3 servers, out of the ~18000 other servers that have statistics enabled for EssentialsX.

I don't recall ever seeing a request for this before (out of the roughly 700 EssentialsX issues I've seen), and nobody else replied indicating that they were in favour of adding this, so I made the personal judgement that this feature is outside the scope of EssentialsX. The other devs are free to reopen this if they feel that it *is* within the plugin's scope, and anyone is free to fork EssentialsX and add the feature themselves should they wish to.

TL;DR If you don't have the time or effort to read my response, I don't have the time or effort to continue this thread.

**Comment by mbax** [comment]
[impmallet](https://github.com/EssentialsX/Essentials/issues/1822#issuecomment-362914234) commented:
>I can assure you that it worked on prior versions, and have evidence to
>prove it.
>
>I looked over your git history and it seems you have a habit of just rapid
>fire closing issues. You personally do not feel this feature has value, but
>you have a user of your plugin that is literally vouching for it.
>
>Thank you for showing me how your dev team operates, I will use a different
>set of plugins going forwars and will stop recommending essentialsx to
>other operators.

@impmallet Rapid fire closing? He dug through the history of the project and proved that the disallowing of nickname duplication has been in the project for seven years! 

Please, provide the evidence you have that will prove this untouched-for-seven-years code wrong. I am genuinely interested in seeing it.

In the mean time, you have clearly demonstrated immaturity which is greatly disappointing to see in this community.

**Comment by LaxWasHere** [comment]
> ☑ Tyrannosaurus Rekt

**Comment by soshimee** [comment]
We really need this :(
